template-for-repository/proofs/Makefile.common

```diff
@@ -119,7 +119,8 @@ PROOFDIR ?= $(abspath .)
 CBMCDIR ?= $(PROOF_ROOT)/cbmc
 
 # Default CBMC flags used for property checking and coverage checking.
-CBMCFLAGS += --unwind 1 $(CBMC_UNWINDSET) --flush
+UNWIND_COUNT = 1
```

Comment:
Shouldn't this be `?=` like the other variables so that it can be overridden in each project or in particular proofs?

Reply:
Hi Felipe, Kareem,

This was hardcoded and not exposed before, so I didn't want to expose it in this PR. But thinking more about whether this should be exposed to end users, I think it shouldn't be, for several reasons:

1. `UNWINDSET` is exposed for users to indicate unwinding counts on individual loops. This is just the default `UNWIND_COUNT` for all the remaining loops that are not in the `UNWINDSET`.  
Setting it to a small value makes CBMC fail fast when the user forgets to add a particular to `UNWINDSET`. Users might potentially shoot themselves in the foot if they start tweaking this default value instead -- proofs might take way too long only to fail on an unwinding assertion violation.
2. Suppose the user sets this to `1`, but if they are also using loop contracts, then we would override their choice and set it to `2`, or we would error out saying that loop contracts require `UNWIND_COUNT` of at least 2.

I'm happy to change to `=` to `?=`, but I don't yet see significant value in exposing this. The `UNWINDSET` is what the users should tweak.

What are your thoughts?

Reply:
Could we have
```
UNWIND ?= --unwind 1
```
just as we do for other CBMC flags like `--bounds-check`, and then later have
```
$(eval UNWIND=--unwind 2)
```
It has always been a slight inconvenience that the starter kit sets `--unwind 1` because this overrides CBMC's ability to figure out on its own how many times to unroll a loop with constant bounds.